reframe/frontend/cli.py

```diff
@@ -562,20 +566,26 @@ def main():
             exec_policy.skip_sanity_check = options.skip_sanity_check
             exec_policy.skip_performance_check = options.skip_performance_check
             exec_policy.keep_stage_files = options.keep_stage_files
+
+            if options.flex_alloc_tasks:
+                printer.warning('--flex-alloc-tasks is deprecated, you should '
+                                'use --flex-alloc-nodes instead: Exiting...')
+                sys.exit(1)
```

Comment:
This warning is given too late. It should be issued before any action, so that it is shown even with the `-l` option.

Reply:
There is a conceptual problem here. We cannot issue a warning and exit. It should be an error, but deprecations are not errors. I suggest the following. Make the default value of both options `None` so that you know if they are specified (but don't change the documentation or the help message). You can then do sth like this:

```python
if options.flex_alloc_tasks:
    printer.warning('deprecation message')
    options.flex_alloc_nodes = options.flex_alloc_nodes or options.flex_alloc_tasks

options.flex_alloc_nodes = options.flex_alloc_nodes or 'idle'
```